### PR TITLE
Replace std::array<double, 3/4> with MathArray.

### PR DESCRIPTION
**Pull request by geektoni**
This PR replaces all the occurrences of `std::array<double, 3>` and `std::array<double, 4>` with the new `Double3` and `Double4` types (aka `MathArray<double, 3>` and `MathArray<double, 4>`)

### CODE REVIEW

File: src/core/sim_object/cell.h [hunk 1]
```diff
@@ -103,10 +103,9 @@ class Cell : public SimObject {
 
       // position
       auto mother_pos = mother_cell->GetPosition();
-      std::array<double, 3> new_position{
-          mother_pos[0] + d_2 * axis_of_division[0],
-          mother_pos[1] + d_2 * axis_of_division[1],
-          mother_pos[2] + d_2 * axis_of_division[2]};
+      Double3 new_position{mother_pos[0] + d_2 * axis_of_division[0],
+                           mother_pos[1] + d_2 * axis_of_division[1],
+                           mother_pos[2] + d_2 * axis_of_division[2]};
```
Comment (breitwieserCern) [inline src/core/sim_object/cell.h]: `auto new_position = mother_pos + d_2 * axis_of_division;`

File: src/core/sim_object/cell.h [hunk 2]
```diff
@@ -358,32 +351,28 @@ class Cell : public SimObject {
...
-        movement_at_next_step[0] =
-            norm[0] * param->simulation_max_displacement_;
-        movement_at_next_step[1] =
-            norm[1] * param->simulation_max_displacement_;
-        movement_at_next_step[2] =
-            norm[2] * param->simulation_max_displacement_;
+        movement_at_next_step.Normalize();
+        movement_at_next_step[0] *= param->simulation_max_displacement_;
+        movement_at_next_step[1] *= param->simulation_max_displacement_;
+        movement_at_next_step[2] *= param->simulation_max_displacement_;
```
Comment (breitwieserCern) [inline src/core/sim_object/cell.h]: `movement_at_next_step *= param->simulation_max_displacement_;`

File: src/core/util/math.h [hunk 3]
```diff
@@ -168,16 +83,14 @@ struct Math {
...
-    auto temp_2 =
-        Math::ScalarMult(std::cos(-theta), Math::Subtract(vector, temp_1));
-    auto temp_3 =
-        Math::ScalarMult(std::sin(-theta), CrossProduct(vector, naxis));
+    auto temp_1 = naxis * (vector * naxis);
+    auto temp_2 = (vector - temp_1) * std::cos(-theta);
+    auto temp_3 = vector.CrossProduct(naxis) * std::sin(-theta);
 
     return {
         temp_1[0] + temp_2[0] + temp_3[0], temp_1[1] + temp_2[1] + temp_3[1],
```
Comment (breitwieserCern) [inline src/core/util/math.h]: `return temp_1 + temp_2 + temp_3`

File: src/core/util/timing_aggregator.h [hunk 4]
```diff
@@ -54,7 +54,9 @@ inline std::ostream& operator<<(std::ostream& os, const TimingAggregator& ta) {
   if (ta.timings_.size() != 0) {
     os << "\033[1mTotal execution time per operation:\033[0m" << std::endl;
     for (auto& timing : ta.timings_) {
-      os << timing.first << ": " << Math::Sum(timing.second) << std::endl;
+      os << timing.first << ": "
+         << std::accumulate(timing.second.begin(), timing.second.end(), 0)
```
Comment (breitwieserCern) [inline src/core/util/timing_aggregator.h]: `<< timing.Sum() `

**Comment by breitwieserCern** [comment]
The following statements can be simplified too.
https://github.com/geektoni/biodynamo/blob/feature/replace_array_with_double/src/core/sim_object/cell.h#L113-L115
https://github.com/geektoni/biodynamo/blob/feature/replace_array_with_double/src/core/sim_object/cell.h#L263-L265
https://github.com/geektoni/biodynamo/blob/feature/replace_array_with_double/src/core/sim_object/cell.h#L316-L318

**Comment by geektoni** [comment]
> The following statements can be simplified too.
> https://github.com/geektoni/biodynamo/blob/feature/replace_array_with_double/src/core/sim_object/cell.h#L113-L115
> https://github.com/geektoni/biodynamo/blob/feature/replace_array_with_double/src/core/sim_object/cell.h#L263-L265
> https://github.com/geektoni/biodynamo/blob/feature/replace_array_with_double/src/core/sim_object/cell.h#L316-L318

The last one cannot be simplified since it is an operation between `Double3` and `Double4` (thus they have incompatible sizes).

**Pull request merged**